static/app/views/starfish/utils/getDateFilters.tsx

```diff
@@ -5,12 +5,10 @@ import {PageFilters} from 'sentry/types';
 export const PERIOD_REGEX = /^(\d+)([h,d])$/;
 export const DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';
 
-export function getDateFilters(selection: PageFilters) {
-  const [_, num, unit] = selection.datetime.period?.match(PERIOD_REGEX) ?? [];
+export function getDateFilters(selection: Partial<PageFilters['datetime']>) {
```

Comment:
Hello!

**Re why do this:** Starfish currently limits date range selection to 7 days, due to performance issues. If your current date filter is set to anything longer than 7 days, when you open Starfish it will be reset. This creates a discontinuity between Starfish and other pages. I talked with @doralchan about this and we think that the date filter's value inside Starfish should be saved separately and in isolation from the global value. This way, when you navigate out of and then back into Starfish, you can resume your previous Starfish workflow with the saved Starfish date range. This is only a special accommodation given the 7-day constraint. Ideally we shouldn't have to do this, and the global filter value should be used.

**Re your namespace suggestion:** that does look nicer. I decided to write these as one-off, Starfish-specific utils because currently I only see them being used by Starfish (Dashboards have a similar mechanism, but they store filter values in the backend not local storage). I'm a little hesitant to incorporate this into `persistence.tsx` & `<PageFiltersContainer />` because doing so would make it much easier to create local filtering contexts, thereby encouraging people to do it. I think local filtering contexts should be limited to when they're absolutely necessary. Here in Starfish it's justified given the 7-day constraint on the date filter. Wdyt?

Reply:
Ah gotcha, thanks for clarifying! We actually [removed the 7-day limit just yesterday](https://github.com/getsentry/sentry/pull/52640), does that mean we don't need this separation anymore?

Reply:
Omg lol yes 😆 Lemme close the PR then